### ISSUE TITLE
When using custom VM Images (UseAgentCustomImage, UseMasterCustomImage), it should possible to provide the subscription ID.

### ISSUE THREAD

**Issue by AgelM** [Azure/aks-engine#721]
**Is this a request for help?**:
No
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
FEATURE REQUEST
---

**What version of acs-engine?**:
v0.21.2
---

Having the ability to use custom VM images provides more possibilities to enhance the cluster. 
The current solution requires to use a Custom Image from the same subscription as where the cluster is to be deployed.
```
"masterProfile":  {
    ...
    "imageReference": {
        "name": "sourceImageName",
        "resourceGroup": "sourceResourceGroup"
    },
    ...
}
``` 
It should be possible to specify the *subscription* where the Image is in order to use an image from a different subscription, without having to copy it, as described below:

```
"masterProfile":  {
    ...
    "imageReference": {
        "name": "sourceImageName",
        "resourceGroup": "sourceResourceGroup",
        "subscriptionId": "sourceSubscriptionID" 
    },
    ...
}
``` 
The documentation should specify that the `servicePrincipal` given in the template should have read access to the subscription with the image or to the image itself.

**Comment by alexeldeib** [comment]
I took a stab at implementing this. I don't think it's possible? Even though the client service principal has access to both subs, ARM won't accept using an image across subscriptions. Error below.

I found Shared Image Galleries, which would seem to work for the same purpose. It would require specifying additional parameters for galleryName, galleryImage, and imageVersion. Or alternatively, allow the full ID.
```
{"code":"BadRequest","message":"{\r\n  \"error\": {\r\n    \"code\": \"BadRequest\",\r\n    \"message\": \"Image sharing not supported for subscription.\"\r\n  }\r\n}"}
```

**Comment by AgelM** [comment]
Well at the time I reported this, I was able to use images from a different subscription within the ARM, given that the image was within the same Region/location as where the deployment is to take place. The Shared Image Gallery was not yet available as a offering that one could use for private Custom Images.

As for your proposal, I believe that having the possibility to use images from the Gallery covers the requirement expressed within this ticket and we should work on integrating that within the Aks-engine.
